core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -0,0 +1,95 @@
+package com.sap.ai.sdk.core;
+
+import static com.sap.ai.sdk.core.AiCoreServiceWithDeployment.getDeploymentId;
+import static com.sap.ai.sdk.core.AiCoreServiceWithDeployment.isDeploymentOfModel;
+
+import com.sap.ai.sdk.core.client.model.AiDeployment;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.exception.DestinationAccessException;
+import com.sap.cloud.sdk.cloudplatform.connectivity.exception.DestinationNotFoundException;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/** Connectivity convenience methods for AI Core. */
+@Slf4j
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class AiCoreService implements AiCoreServiceStub {
+
+  // the destination to be used for AI Core service calls.
+  @Nonnull private final Supplier<Destination> destination;
+
+  /** Create a new instance of the AI Core service. */
+  public AiCoreService() {
+    this(AiCoreService::getDefaultDestination);
+  }
+
+  @Nonnull
+  @Override
+  public Destination destination() {
+    return destination.get();
+  }
+
+  /**
+   * Set a specific base destination.
+   *
+   * @param destination The destination to be used for AI Core service calls.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public AiCoreService withDestination(@Nonnull final Destination destination) {
```

Comment:
I would rather have:
```diff
-withDestination()
+forDestination()
```
```diff
-withDeploymentByModel()
+forDeploymentWithModel()   or forModel()...
```
```diff
-withDeploymentByScenario()
+forDeploymentWithScenario()
```

Reply:
While I understand and mostly agree, the "wither" pattern is [established](https://projectlombok.org/api/lombok/With) in Java world - I think.

Maybe we can avoid mixing "with" and "for" method prefix(?)

Reply:
I think the "wither" pattern is typically used for changing an existing instance, where `for` is more a static factory prefix. So to me `withDestination()` would be fine, but I'd probably look for something else instead of e.g.  `withDeploymentByScenario()` . See also [this comment](https://github.com/SAP/ai-sdk-java/pull/79/files?diff=unified&w=1#r1782942027).

Reply:
with means in addition. What we are doing is replacing the original URL, and creating a `new AiCoreService()`. which is kind of a static factory.